source/common/network/resolver_impl.cc

```diff
@@ -48,9 +48,20 @@ InstanceConstSharedPtr resolveProtoAddress(const envoy::config::core::v3::Addres
   case envoy::config::core::v3::Address::AddressCase::kSocketAddress:
     return resolveProtoSocketAddress(address.socket_address());
   case envoy::config::core::v3::Address::AddressCase::kPipe:
-    return InstanceConstSharedPtr{new PipeInstance(address.pipe().path())};
+    return std::make_shared<PipeInstance>(address.pipe().path());
+  case envoy::config::core::v3::Address::AddressCase::kEnvoyInternalAddress:
+    switch (address.envoy_internal_address().address_name_specifier_case()) {
+    case envoy::config::core::v3::EnvoyInternalAddress::AddressNameSpecifierCase::
+        kServerListenerName:
+      return std::make_shared<EnvoyInternalInstance>(
+          address.envoy_internal_address().server_listener_name());
+    default:
```

Comment:
This is verified by schema and can't happen. Replace the default with the NOT_SET variant to make it compile and then use NOT_REACHED. The same should be done below.